docs/data/material/components/autocomplete/Asynchronous.js

```diff
@@ -14,44 +14,30 @@ function sleep(duration) {
 export default function Asynchronous() {
   const [open, setOpen] = React.useState(false);
   const [options, setOptions] = React.useState([]);
-  const loading = open && options.length === 0;
-
-  React.useEffect(() => {
-    let active = true;
-
-    if (!loading) {
-      return undefined;
-    }
```

Comment:
I also refactored the code to remove two useEffect hooks, achieving the same functionality with more readable code.